bigtop-deploy/puppet/modules/gpdb/manifests/init.pp

```diff
@@ -0,0 +1,440 @@
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+class gpdb {
+
+
+  class deploy($roles) {
+    if ("gpdb-master" in $roles or "gpdb-segment" in $roles) {
+      include gpdb::common
+    }
+  }
+
+
+  class common($nodes, $gp_home, $db_base_dir, $master_db_port, $segment_db_port_prefix) {
+
+    include stdlib
+
+    class { 'gpdb::common::install_packages': }
+
+    class { 'gpdb::common::gpadmin_user':}
+
+    class { 'gpdb::common::prepare_file_structure':
+      base_dir => "$db_base_dir",
+      require  => Class['gpdb::common::gpadmin_user']
+    }
+
+    class { 'gpdb::common::prepare_db_dirs':
+      base_dir => "$db_base_dir",
+      nodes    => $nodes,
+      gp_home  => $gp_home,
+      master_port => $master_db_port,
+      segment_port_prefix => $segment_db_port_prefix,
+      require  => [Class['gpdb::common::prepare_file_structure'], Package["gpdb"]]
+    }
+
+    gpdb::server{"stop_if_runned":
+      nodes => $gpdb::common::nodes,
+      gp_home => $gpdb::common::gp_home,
+      db_base_dir => $gpdb::common::db_base_dir,
+      master_db_port => $gpdb::common::master_db_port,
+      segment_db_port_prefix => $gpdb::common::segment_db_port_prefix,
+      require  => Class['gpdb::common::prepare_db_dirs'],
+      start_or_stop => stopped,
+    }
+
+
+    class { 'gpdb::common::configure_master_node':
+      base_dir => "$db_base_dir",
+      nodes    => $nodes,
+      gp_home  => $gp_home,
+      master_port => $master_db_port,
+      segment_port_prefix => $segment_db_port_prefix,
+      require  => Class['gpdb::common::prepare_db_dirs']
+    }
+
+    class { 'gpdb::common::stop_master_in_admin_mode':
+      base_dir => "$db_base_dir",
+      gp_home  => $gp_home,
+      master_port => $master_db_port,
+      require  => Class['gpdb::common::configure_master_node']
+    }
+
+    gpdb::server{"start":
+      nodes => $gpdb::common::nodes,
+      gp_home => $gpdb::common::gp_home,
+      db_base_dir => $gpdb::common::db_base_dir,
+      master_db_port => $gpdb::common::master_db_port,
+      segment_db_port_prefix => $gpdb::common::segment_db_port_prefix,
+      require  => Class['gpdb::common::stop_master_in_admin_mode'],
+      start_or_stop => running,
+    }
+
+
+
+    class gpadmin_user{
+      group { 'gpadmin':
+        ensure => 'present',
+      }
+
+      user { 'gpadmin':
+        ensure            => 'present',
+        system            => false,
+        managehome        => true,
+        groups            => 'gpadmin',
+        shell             => '/bin/bash',
+      }
+    }
+
+    class install_packages{
```

Comment:
Also, I suggest we add these packages to the Bigtop's deploy docker image in a separate JIRA.